### ISSUE TITLE
Encoding error during delivery

### ISSUE THREAD

**Issue by gdchamal** [CaliOpen/Caliopen#1375]
Found in production now :

Invalid file name encoding:
```
[mq-worker-d996c9fd-nl55v] DataError: {"attachments": {"1": {"file_name": ["Invalid UTF-8 data."]}}} 
[mq-worker-d996c9fd-nl55v] ERROR:subscribers:deliver process failed for raw {u'message_id': u'29501724-ff0c-463c-8874-d059647d86d8', u'user_id': u'ba659d8e-3f96-485e-a8f4-2f873787bf66', u'order': u'process_raw', u'identity_id': u'974e91c4-ab04-404f-9ded-ca3006ec7e99'}: None 
```

Invalid body encoding:
```
[mq-worker-d996c9fd-5rcl4] Traceback (most recent call last): 
[mq-worker-d996c9fd-5rcl4]   File "/srv/caliopen/src/backend/interfaces/NATS/py.client/caliopen_nats/subscribers.py", line 46, in process_raw 
[mq-worker-d996c9fd-5rcl4]     new_message = deliver.process_raw(payload['message_id']) 
[mq-worker-d996c9fd-5rcl4]   File "/srv/caliopen/src/backend/interfaces/NATS/py.client/caliopen_nats/delivery.py", line 37, in process_raw 
[mq-worker-d996c9fd-5rcl4]     message = self.qualifier.process_inbound(raw) 
[mq-worker-d996c9fd-5rcl4]   File "/srv/caliopen/src/backend/components/py.pi/caliopen_pi/qualifiers/mail.py", line 88, in process_inbound 
[mq-worker-d996c9fd-5rcl4]     email = MailMessage(raw.raw_data) 
[mq-worker-d996c9fd-5rcl4]   File "/srv/caliopen/src/backend/main/py.main/caliopen_main/message/parsers/mail.py", line 143, in __init__ 
[mq-worker-d996c9fd-5rcl4]     self.get_bodies() 
[mq-worker-d996c9fd-5rcl4]   File "/srv/caliopen/src/backend/main/py.main/caliopen_main/message/parsers/mail.py", line 183, in get_bodies 
[mq-worker-d996c9fd-5rcl4]     self.body_plain = to_utf8(body_plain, charset) 
[mq-worker-d996c9fd-5rcl4]   File "/srv/caliopen/src/backend/main/py.main/caliopen_main/common/helpers/strings.py", line 34, in to_utf8 
[mq-worker-d996c9fd-5rcl4]     "with error : {}".format(input, exc)) 
[mq-worker-d996c9fd-5rcl4] UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 40: ordinal not in range(128) 
[mq-worker-d996c9fd-5rcl4] ERROR:subscribers:deliver process failed for raw {u'message_id': u'98f67f8f-5b85-4c27-bf5b-ce58e0ab62c8', u'user_id': u'ba659d8e-3f96-485e-a8f4-2f873787bf66', u'order': u'process_raw', u'identity_id': u'974e91c4-ab04-404f-9ded-ca3006ec7e99'}: None 
```

Invalid subject encoding:
```
[mq-worker-d996c9fd-x56h2]   File "/srv/caliopen/src/backend/interfaces/NATS/py.client/caliopen_nats/subscribers.py", line 46, in process_raw 
[mq-worker-d996c9fd-x56h2]     new_message = deliver.process_raw(payload['message_id']) 
[mq-worker-d996c9fd-x56h2]   File "/srv/caliopen/src/backend/interfaces/NATS/py.client/caliopen_nats/delivery.py", line 37, in process_raw 
[mq-worker-d996c9fd-x56h2]     message = self.qualifier.process_inbound(raw) 
[mq-worker-d996c9fd-x56h2]   File "/srv/caliopen/src/backend/components/py.pi/caliopen_pi/qualifiers/mail.py", line 160, in process_inbound 
[mq-worker-d996c9fd-x56h2]     raise exc 
[mq-worker-d996c9fd-x56h2] DataError: {"subject": ["Invalid UTF-8 data."]} 
[mq-worker-d996c9fd-x56h2] ERROR:subscribers:deliver process failed for raw {u'message_id': u'5ae8db43-b770-4f1e-b61f-0ac93cec2536', u'user_id': u'f365fc36-3401-4ee3-95cb-9c6f6078f00d', u'order': u'process_raw', u'identity_id': u'2ed4970b-75bc-496d-9842-17a69fed643b'}: None 
```

**Comment by gdchamal** [comment]
Parse address the return :
```
[mq-worker-d996c9fd-t9mzq]     participant = MailParticipant(participant_type, addr.lower()) 
[mq-worker-d996c9fd-t9mzq]   File "/srv/caliopen/src/backend/main/py.main/caliopen_main/message/parsers/mail.py", line 109, in __init__ 
[mq-worker-d996c9fd-t9mzq]     parts = clean_email_address(addr) 
[mq-worker-d996c9fd-t9mzq]   File "/srv/caliopen/src/backend/main/py.main/caliopen_main/common/helpers/normalize.py", line 13, in clean_email_address 
[mq-worker-d996c9fd-t9mzq]     real_name, email = parseaddr(addr.replace('\r', '')) 
[mq-worker-d996c9fd-t9mzq] UnicodeDecodeError: 'ascii' codec can't decode byte 0xe9 in position 25: ordinal not in range(128) 
[mq-worker-d996c9fd-t9mzq] ERROR:subscribers:deliver process failed for raw {u'message_id': u'258bb16b-2ae1-47e9-a706-62d41f84a338', u'user_id': u'ba659d8e-3f96-485e-a8f4-2f873787bf66', u'order': u'process_raw', u'identity_id': u'974e91c4-ab04-404f-9ded-ca3006ec7e99'}: None 
```

**Comment by gdchamal** [comment]
all problems are addressed by #1380
